src/main/java/org/terasology/signalling/componentSystem/SignalSystem.java

```diff
@@ -273,6 +296,11 @@ private void outputSignalToSimpleConsumer(EntityRef entity, boolean result) {
         }
     }
 
+    /**
+     * Detects a XOR signal based on the sides recieving a signal
```

Comment:
just mention this is for the XOR block and same thing with the other gate signals.